### PR TITLE
feat: fix notifications and manage them

### PR DESCRIPTION
**Pull request by aesedepece**
Ref #953

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -34,16 +36,26 @@ class ApiClient {
     return this.ws.subscribe(method)
   }
 
-  async unsubscribe(method) {
-    return this.ws.unsubscribe(method)
+  async unsubscribe(event) {
```
Comment (Tommytrg) [inline src/api.js:39]: Maybe we could delete `subscribe` and `unsubscribe` methods.

File: src/api.js [hunk 2]
```diff
@@ -70,13 +82,17 @@ export class WalletApi {
         .catch(this._handleError)
   }
 
+  // This is overriding the native `client.subscribe` because it lacks support for `params`
   async subscribeToNotifications(params, cb) {
-    return this._callApiMethod('subscribe_notifications')([params.sessionId])
+    return this._callApiMethod('rpc.on')([params.session_id])
       .then(_ => {
         this.client.on('notifications', cb)
```
Comment (Tommytrg) [inline src/api.js:89]: As I ask below, what do you think about this?
```suggestion
        this.client.on('notifications', ([notification]) => cb(notification))
```

File: src/store/wallet.js [hunk 3]
```diff
@@ -483,14 +482,20 @@ export default {
...
     },
     subscribeToWalletNotifications: async function(context) {
-      context.state.api.subscribeToNotifications(
+      await context.state.api.subscribeToNotifications(
         { session_id: this.state.wallet.sessionId },
-        notification =>
-          context.commit('setBalance', {
-            balance: { total: notification[0].accountBalance.amount },
-          })
+        ([notification]) => {
```
Comment (Tommytrg) [inline src/store/wallet.js:487]: What do you think about  add a callback to flatten [notification] i in `subscribeToNotifications` to simplify the api? I add a suggestion for that in `subscribeToWalletNotifications`.

File: src/store/wallet.js [hunk 4]
```diff
@@ -483,14 +482,20 @@ export default {
       }
     },
     subscribeToWalletNotifications: async function(context) {
-      context.state.api.subscribeToNotifications(
+      await context.state.api.subscribeToNotifications(
```
[the remark this answers is not in the document]
Comment (aesedepece) [reply src/store/wallet.js:485]: Don't really know it. I think my IDE complaint about the lack thereof.

**Comment by aesedepece** [comment]
This is superseded by #1067

**Pull request closed without merging**